### PR TITLE
[zh-cn] Update security/SRI

### PR DESCRIPTION
**Pull request by jasonren0403**

### CODE REVIEW

File: files/zh-cn/web/security/subresource_integrity/index.md [hunk 1]
```diff
@@ -1,94 +1,110 @@
...
+使用{{Glossary("CDN", "内容分发网络（CDN）")}}在多个站点之间共享脚本和样式表等文件可以提高站点性能并节省带宽。然而，使用 CDN 也存在风险，如果攻击者获得对 CDN 的控制权，则可以将任意恶意内容注入到 CDN 上的文件中（或完全替换掉文件），因此可能潜在地攻击所有从该 CDN 获取文件的站点。
 
-> **备注：** SRI 并不能规避所有的风险。第三方库经常会自己请求额外的信息，这就有可能会携带用户的账号密码等关键信息。这些经常需要 js 功能的支持，比如一个地图库会需要取\<svg>数据来渲染，但是包含点击事件。
+子资源完整性使你能够减轻这种攻击的一些风险，确保你的网络应用程序或网络文档（从 CDN 或任何地方）获取的文件在交付时没有被第三方注入任何额外的内容，也没有对这些文件进行任何其他形式的修改。
 
-## 如何使用 SRI
+## SRI 如何使用
 
-将使用 base64 编码过后的文件哈希值写入你所引用的 {{HTMLElement("script")}} 或 {{HTMLElement("link")}} 标签的 **integrity** 属性值中即可启用子资源完整性功能。
+使用子资源完整性功能的方法是，在任何 {{HTMLElement("script")}} 或 {{HTMLElement("link")}} 元素的  `integrity` 属性值中，指定你要告诉浏览器所获取的资源（或文件）的 base64 编码的加密哈希值。
```
Comment (jasonren0403) [inline files/zh-cn/web/security/subresource_integrity/index.md:18]:
```suggestion
使用子资源完整性功能的方法是，在任何 {{HTMLElement("script")}} 或 {{HTMLElement("link")}} 元素的 `integrity` 属性值中，指定你要告诉浏览器所获取的资源（或文件）的 base64 编码的加密哈希值。
```

File: files/zh-cn/web/security/subresource_integrity/index.md [hunk 2]
```diff
@@ -1,94 +1,110 @@
...
 
 ## 浏览器如何处理 SRI
 
 浏览器根据以下步骤处理 SRI：
 
-1. 当浏览器在 {{HTMLElement("script")}} 或者 {{HTMLElement("link")}} 标签中遇到 **integrity** 属性之后，会在执行脚本或者应用样式表之前对比所加载文件的哈希值和期望的哈希值。
-2. 当脚本或者样式表的哈希值和期望的不一致时，浏览器必须拒绝执行脚本或者应用样式表，并且必须返回一个网络错误说明获得脚本或样式表失败。
+1. 当浏览器在 {{HTMLElement("script")}} 或者 {{HTMLElement("link")}} 标签中遇到 `integrity` 属性之后，会在执行脚本或者应用样式表之前对比所加载文件的哈希值和期望的哈希值。
+
+    对于从其他来源提供的资源的子资源完整性验证，浏览器还使用[跨源资源共享（CORS）]（/zh-CN/docs/Web/HTTP/CORS）检查资源，以确保提供资源的来源允许它与请求来源共享。
```
Comment (jasonren0403) [inline files/zh-cn/web/security/subresource_integrity/index.md:105]:
```suggestion
    对于从其他来源提供的资源的子资源完整性验证，浏览器还使用[跨源资源共享（CORS）](/zh-CN/docs/Web/HTTP/CORS)检查资源，以确保提供资源的来源允许它与请求来源共享。
```

File: files/zh-cn/web/security/subresource_integrity/index.md [hunk 3]
```diff
@@ -100,8 +116,10 @@ Content-Security-Policy: require-sri-for style;
 
 ## 相关资料
 
-- Content Security Policy 内容安全策略
-- {{httpheader("Content-Security-Policy")}}
-- [A CDN that can not XSS you: Using Subresource Integrity](https://frederik-braun.com/using-subresource-integrity.html)
+- [内容安全策略](/zh-CN/docs/Web/HTTP/CSP)
+- HTTP 标头 {{httpheader("Content-Security-Policy")}}
+- [一个不能对你进行 XSS 的 CDN：使用 SRI](https://frederik-braun.com/using-subresource-integrity.html)
+- [来自W3C的 SRI 测试](https://w3c-test.org/subresource-integrity/subresource-integrity.html)
```
Comment (fwqaaq) [inline files/zh-cn/web/security/subresource_integrity/index.md:122]:
```suggestion
- [来自 W3C 的 SRI 测试](https://w3c-test.org/subresource-integrity/subresource-integrity.html)
```

File: files/zh-cn/web/security/subresource_integrity/index.md [hunk 4]
```diff
@@ -100,8 +116,10 @@ Content-Security-Policy: require-sri-for style;
...
+- [SRI Hash 生成器](https://www.srihash.org/)
 
 {{QuickLinksWithSubpages("/zh-CN/docs/Web/Security")}}
```
Comment (yin1999) [inline files/zh-cn/web/security/subresource_integrity/index.md:125]: 请将其挪到开头，和英文文档保持一致

**Review by yin1999 — APPROVED** [review]

**Pull request merged**